### PR TITLE
fix(interactive-shell): forward status through StreamingConsole(output=)

### LINKED ISSUE #4657: Q-46: Forward Rich status through StreamingConsole(output=)
## Summary
`StreamingConsole` (`surfaces/interactive_shell/ui/streaming/console.py`) already forwards `print` / `use_theme` when `output=` is set (see `tests/interactive_shell/ui/test_streaming_console_output.py`). Loaders and slash commands call `console.status(...)` (e.g. `surfaces/interactive_shell/ui/components/loaders.py`) but `status` is not yet forwarded to the injected console. Forward `status` the same way `print`/`use_theme` are forwarded, and add a unit test next to the existing output tests.

## Done when
- [ ] `status` calls are forwarded to the injected console when `output=` is set
- [ ] New unit test added alongside existing output tests
- [ ] lint/typecheck pass

## Files
`surfaces/interactive_shell/ui/streaming/console.py` + `tests/interactive_shell/ui/test_streaming_console_output.py`

### PR DESCRIPTION
**Pull request by arulgoyal**
Loaders and slash commands call console.status(...) as a context manager, but it wasn't forwarded to an injected output console the way print/use_theme already are — spinners rendered on the wrong console when output= was set. Mirrors the existing use_theme() forwarding pattern.

Closes #4657

Fixes #4657

#### Describe the changes you have made in this PR -
Added a `status()` override to `StreamingConsole` (`surfaces/interactive_shell/ui/streaming/console.py`) that forwards to `self._output.status(...)` when an `output=` console is injected, falling back to `super().status(...)` otherwise — the same pattern already used by `use_theme()` and `print()` on this class. Added `test_streaming_console_status_forwards_to_output` alongside the existing forwarding tests in `tests/interactive_shell/ui/test_streaming_console_output.py`, asserting the returned `Status`'s `.console` is the injected target console.

### Demo/Screenshot for feature changes and bug fixes -
```
tests/interactive_shell/ui/test_streaming_console_output.py::test_streaming_console_with_output_skips_tty_prep PASSED
tests/interactive_shell/ui/test_streaming_console_output.py::test_streaming_console_without_output_still_prepares_tty PASSED
tests/interactive_shell/ui/test_streaming_console_output.py::test_streaming_console_use_theme_forwards_to_output PASSED
tests/interactive_shell/ui/test_streaming_console_output.py::test_streaming_console_status_forwards_to_output PASSED
============================ 4 passed in 2.18s ============================
```

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
`StreamingConsole` wraps Rich's `Console` and optionally forwards to an injected `output` console (used when embedding the shell's output into another surface, e.g. a capture/replay path). `print` and `use_theme` already had this forwarding; `status` — used by loaders and slash commands as a `with console.status(...):` spinner — did not, so spinners rendered against the wrong console whenever `output=` was set. I mirrored `use_theme()`'s approach rather than `print()`'s, because Rich's `Console.status()` has a fixed, typed signature (`status`, `spinner`, `spinner_style`, `speed`, `refresh_per_second`) rather than `print`'s open `*args/**kwargs`, so a typed override matches the real call surface and lets mypy check it. No wrapping is needed on the return value — Rich's `Status` context manager is already bound to whichever console built it, so returning `self._output.status(...)` directly is correct. Alternative considered: reusing `print()`'s pattern with `*args/**kwargs` — rejected since it would lose type-checking on `status()`'s kwargs and diverge from the `use_theme` precedent already established in this file. Edge case covered: the no-`output` path is untouched, falling through to `super().status(...)` exactly as before, so REPL behavior when nothing is injected is unaffected (covered by the existing print/use_theme tests plus manual read of the diff).

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by arulgoyal** [comment]
@greptile review

**Review by cerencamkiran — COMMENTED** [review]
Looks good. Thanks for your time.

**Comment by Devesh36** [comment]
Hey fix the failing test please

**Comment by arulgoyal** [comment]
> Hey fix the failing test please

sure will review this and share the updated PR soon !

**Comment by arulgoyal** [comment]
Not a real test failure  `test (e2e-general)` died in the `Install uv` setup step (`fetch failed` pulling uv's version manifest from raw.githubusercontent.com), before any tests ran. Ran the same suite locally: 159 passed, 37 skipped, 0 failed. Looks like a transient CI network blip  can we re-run the job?.

**Comment by arulgoyal** [comment]
@Devesh36 Please check this !

File: surfaces/interactive_shell/ui/streaming/console.py [hunk 1]
```diff
@@ -84,6 +86,32 @@ def use_theme(self, theme: Theme, *, inherit: bool = True) -> ThemeContext:
...
+                speed=speed,
+                refresh_per_second=refresh_per_second,
+            )
+        return super().status(
+            status,
+            spinner=spinner,
+            spinner_style=spinner_style,
+            speed=speed,
+            refresh_per_second=refresh_per_second,
+        )
```
Comment (Davidson3556) [inline surfaces/interactive_shell/ui/streaming/console.py]: Agreed on the typed override over `*args/**kwargs`. You can keep the typing and halve the body:

```python
kwargs = dict(
    spinner=spinner,
    spinner_style=spinner_style,
    speed=speed,
    refresh_per_second=refresh_per_second,
)
if self._output is not None:
    return self._output.status(status, **kwargs)
return super().status(status, **kwargs)
```

Also worth a comment noting the defaults are copied from Rich — if Rich changes `refresh_per_second`, this pins the old value silently.

File: tests/interactive_shell/ui/test_streaming_console_output.py [hunk 2]
```diff
@@ -102,3 +102,20 @@ def test_streaming_console_use_theme_forwards_to_output() -> None:
...
+        _Spinner(),
+        threading.Event(),
+        output=target,
+        force_terminal=False,
+        highlight=False,
+    )
+    status = console.status("working...")
+    assert status.console is target
+    with status:
+        pass
```
Comment (Davidson3556) [inline tests/interactive_shell/ui/test_streaming_console_output.py:168]: The PR body says the no-`output` path is covered by the existing print/use_theme tests, but those don't touch `status`. One more assert pins it:

```python
assert StreamingConsole(_Spinner(), threading.Event()).status("x").console is console
```

**Review by Davidson3556 — COMMENTED** [review]
Great job. `status` is the only method with this bug — `rule` and `input` both render via `self.print`, which this class already forwards.

Two small things inline, neither blocking. Branch is behind main.

**Comment by Davidson3556** [comment]
cl is also failing , kindly fix

**Comment by arulgoyal** [comment]
@greptile review

**Comment by arulgoyal** [comment]
@Devesh36 @Davidson3556 Please check now !! I have addressed the comments !

**Comment by Davidson3556** [comment]
cl is failing, fix it

**Comment by arulgoyal** [comment]
I'll check! It was getting passed but after merging main branch it got failed

**Comment by Davidson3556** [comment]
LGTM
